cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/repo.go

```diff
@@ -47,22 +47,69 @@ var (
 	defaultPollInterval = metav1.Duration{Duration: 10 * time.Minute}
 )
 
-// namespace maybe apiv1.NamespaceAll, in which case repositories from all namespaces are returned
-func (s *Server) listReposInNamespace(ctx context.Context, namespace string) ([]sourcev1.HelmRepository, error) {
-	client, err := s.getClient(ctx, namespace)
+func (s *Server) listAllRepos(ctx context.Context) ([]sourcev1.HelmRepository, error) {
+	// the actual List(...) call will be executed in the context of
+	// kubeapps-internal-kubeappsapis service account
+	// ref https://github.com/kubeapps/kubeapps/issues/4390 for explanation
+	backgroundCtx := context.Background()
```

Comment:
Similar question about the context here - why not pass `ctx` into the function below? Context's are meant to be passed right through the request chain normally (though there are some exceptions).

Reply:
Well, this isn't the right context for the List(..) call. I don't want caller credentials in this call. There maybe some information from the caller ctx that I could use, like timeout, if set, but I didn't want to use it "AS IS"